apps/showcase/src/components/training/training.component.ts

```diff
@@ -0,0 +1,238 @@
+  public title = '';
+
+  private async loadStepContent(step: TrainingStep) {
+    if (!step.dynamicContent.htmlContent()) {
+      void this.loadResource(step.description.htmlContentUrl).then((content) =>
+        step.dynamicContent.htmlContent.set(content)
+      );
+    }
+    const fileConfiguration = step.description.filesConfiguration;
+    if (fileConfiguration?.urls && !step.dynamicContent.project()?.files) {
+      await this.updateStepDynamicContent(step, Object.entries(fileConfiguration.urls).map(([path, url])=> ({path, url})));
+    }
+    if (fileConfiguration?.solutionUrls && !step.dynamicContent.solutionProject()?.files) {
+      await this.updateStepDynamicContent(step, [
+        ...Object.entries(fileConfiguration.urls).map(([path, url]) => ({path, url})),
+        ...Object.entries(fileConfiguration.solutionUrls).map(([path, url]) => ({path, url}))
+      ],
+      true);
+    }
+  }
+
+  private async loadSteps() {
+    const program: TrainingStepDescription[] = JSON.parse(await this.loadResource(`${this.trainingPath}/program.json`));
+    const stepsToLoad: TrainingStep[] = [];
+    program.forEach((desc) => {
+      const step = {
+        description: desc,
+        dynamicContent: {
+          htmlContent: signal(''),
+          project: signal(null),
+          solutionProject: signal(null)
+        }
+      };
+      void this.loadStepContent(step);
+      stepsToLoad.push(step);
+    });
+    this.steps.set(stepsToLoad);
+    const stepParam = window.location.href.match(new RegExp('#([0-9]+)'))?.[1];
+    const stepRequested = !stepParam || Number.isNaN(stepParam) ? 0 : Number(stepParam);
+    this.setCurrentStep(stepRequested);
+  }
+
+  private getFilesContent(resources: Resource[]) {
+    return resources.reduce((fileSystemTree: FileSystemTree, resource) => {
+      if (resource.path === './' || resource.path === '.') {
+        return {...fileSystemTree, ...JSON.parse(resource.content)} as FileSystemTree;
+      }
+      const sanitizedPath = resource.path.replace(new RegExp('^[.]/?'), '');
+      const parsedPath = sanitizedPath.split('/');
+      parsedPath.reduce((pointer, path, index) => {
+        if (path.indexOf('.') > -1) {
+          const parsedContent = JSON.parse(resource.content);
+          pointer[path] = {file: {contents: parsedContent[path].file.contents}} as FileNode;
+          return pointer;
+        } else if (index === parsedPath.length - 1) {
+          // eslint-disable-next-line no-prototype-builtins
+          if (pointer[path]?.hasOwnProperty('file')) {
+            throw new Error('Cannot override a file with a folder');
+          }
+          pointer[path] ||= {directory: {}};
+          const pathDirectory = (pointer[path] as DirectoryNode).directory;
+          const overrides = Object.entries(JSON.parse(resource.content) as {[key: string]: FileNode | DirectoryNode});
+          overrides.forEach(([key, content]) =>
+            // eslint-disable-next-line no-prototype-builtins
```

Comment:
Regarding the story for ESLint configuration 
do we want to keep this [rule](https://eslint.org/docs/latest/rules/no-prototype-builtins) activated ?

Reply:
Let's discuss this with the team

Reply:
Updated by using `Object.prototype.hasOwnProperty.call` so this rule is no longer needed (but we can still discuss this during the team discussion)

Reply:
Conclusion: We will keep this rule activated.